ocaml/xapi/xapi_pool_update.ml

```diff
@@ -166,22 +165,88 @@ exception Missing_update_key of string
 exception Bad_update_info
 exception Invalid_update_uuid of string
 
-let check_unsigned_update_fist path =
+let check_unsigned_update_fist uuid =
   match Xapi_fist.allowed_unsigned_updates () with
   | None -> false
   | Some fist ->
-    let sha1 =
-      Sha1sum.sha1sum (fun checksum_fd ->
-        let (_: int64) = Unixext.with_file path [ Unix.O_RDONLY ] 0 (fun fd ->
-          Unixext.copy_file fd checksum_fd
-        ) in
-        ()
-      )
-    in
-    debug "Patch Sha1sum: %s" sha1;
-    let fist_sha1s = String.split_f String.isspace fist in
+    debug "Pool update uuid: %s" uuid;
+    let fist_uuids = String.split_f String.isspace fist in
```

Comment:
I suspect Matthew Allen will be unhappy about this change. I was wondering if instead we could calculate the sha1sum of the VDI during the introduce call and stick it into the database. We could then keep the same semantics of the fist point. We might also have to verify the sha1sum on attach too though... Matthew is on vacation until next Monday, so we can take it up with him then. I'm happy for this to go in as is (since this is only going onto a branch), so long as we make sure we discuss this with Matthew when he's back.